tests/react/async.test.tsx

```diff
@@ -1143,3 +1144,229 @@ it('multiple derived atoms with dependency chaining and async write (#813)', asy
     getByText('bName: beta')
   })
 })
+
+it('[renderHook] resolves dependencies reliably after a delay', async () => {
+  const countAtom = atom(0)
+  const asyncCountAtom = atom(0)
+
+  const resolve: (() => void)[] = []
+  const asyncAtom = atom(async (get) => {
+    const count = get(countAtom)
+    await new Promise<void>((r) => resolve.push(r))
+    console.log(`resolved (${count})`)
+    return count
+  })
+
+  const derivedAsyncAtom = atom(
+    async (get, { setSelf }) => {
+      const count = get(countAtom)
+      await microtask()
+      console.log(`derived (${count})`)
+      const asyncCount = await get(asyncAtom)
+      console.log(`derived (${count})`, asyncCount)
+      setSelf(asyncCountAtom, asyncCount)
+    },
+    (_get, set, ...args: Parameters<Setter>) => set(...args)
+  )
+
+  const derivedSyncAtom = atom((get) => {
+    get(derivedAsyncAtom)
+  })
+
+  function useTest() {
+    const asyncCount = useAtomValue(asyncCountAtom)
+    useAtom(derivedSyncAtom)
+    const [count, setCount] = useAtom(countAtom)
+    return { count, setCount, asyncCount }
+  }
+
+  const { result } = renderHook(useTest)
+  const { setCount } = result.current
+
+  await microtask()
+
+  expect(result.current.count).toBe(0)
+  expect(result.current.asyncCount).toBe(0)
+
+  await act(() => setCount(increment))
+  await act(() => setCount(increment))
+
+  resolve[0]!()
+  resolve[1]!()
+  resolve[2]!()
+
+  await microtask()
+  await microtask()
+  await microtask()
+  await macrotask()
+
+  expect(result.current.count).toBe(2)
+  expect(result.current.asyncCount).toBe(2)
+
+  await act(() => setCount(increment))
+  await act(() => setCount(increment))
+
+  resolve[3]!()
+  resolve[4]!()
+
+  await microtask()
+  await microtask()
+  await microtask()
+  await macrotask()
+
+  expect(result.current.count).toBe(4)
+  expect(result.current.asyncCount).toBe(4) // 3
+})
+
+it('[render] resolves dependencies reliably after a delay', async () => {
```

Comment:
Sorry, you already have it? Can we remove renderHook tests?

Reply:
sure. I just left it here for posterity, in case we felt it was a more reliable way to test. I'll remove it.